frontend/src/games/memory-game-with-numbers.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState, useEffect } from "react";
+
+const MemoryGameWithNumbers = () => {
+    const [grid, setGrid] = useState<(number | null)[]>([]);
+    const [clickedNumbers, setClickedNumbers] = useState<number[]>([]);
+    const [isGameStarted, setIsGameStarted] = useState(false);
+    const [showNumbers, setShowNumbers] = useState(true);
+    const [result, setResult] = useState<string | null>(null);
+    const [victoryList, setVictoryList] = useState<number[]>([1, 2, 3, 4, 5]);
+    const [maxNumber, setMaxNumber] = useState<number>(5); 
+
+    const startGame = async () => {
+        try {
+            const response = await fetch(`http://localhost:5252/MemoryGameWithNumbers/start?maxNumber=${maxNumber}`);
+            const data = await response.json();
+
+            setGrid(data);
+            setIsGameStarted(true);
+            setClickedNumbers([]); 
+            setShowNumbers(true); 
+
+
+            setTimeout(() => {
+                setShowNumbers(false);
+                setResult("");
+            }, 3000);
+        } catch (error) {
+            console.error("Error starting game:", error);
+        }
+    };
+
+    const handleClick = (num: number | null, index: number) => {
```

Comment:
Šitoj funkcijoj ```index``` nenaudojamas, tai jį reiktų arba išimt visiškai, arba (jei pvz jo reikės ateity) pervadint į ```_index```, kad būtų aišku, jog jis ten ne be reikalo.